modules/EnsEMBL/Web/Component/Shared.pm

```diff
@@ -957,10 +958,10 @@ sub check_for_missing_species {
 
   foreach (keys %{$align_details->{'species'}}) {
     next if $_ eq $species;
-    my $sp_url = $species_defs->production_name_mapping($_);
+    my $sp_url = $url_lookup->{$_};
     if ($align_details->{'class'} !~ /pairwise/
         && ($self->param(sprintf 'species_%d_%s', $align, lc) || 'off') eq 'off') {
-      push @skipped, $_ unless ($args->{ignore} && $args->{ignore} eq 'ancestral_sequences');
+      push @skipped, $sp_url unless ($args->{ignore} && $args->{ignore} eq 'ancestral_sequences');
```

Comment:
dont think you need to change this line, should it still not be `$_` instead of `$sp_url` ? but i vaguely remember that you mention something about this.

Reply:
We need production name here @ens-ma7 because further down on line 978 where we use this `@skipped` to display a warning we call it like this `$species_defs->species_label($_)` So either we keep production name here or we get rid of the `species_label()` call.

Reply:
Jyo is right - the code is correct. We do need the species_label call (to get the final text shown on the web page), and having checked all the compara pages the other day, I made several corrections to fix places where they said "Ancestral species" instead of the species name.